File: catkin_ws/src/rospy_service/scripts/server_str.py
#! /usr/bin/env python3

import rospy

from rospy_service.srv import string 

def callback(req):
    print("Request: ", req.x)
    return "world!!"

rospy.init_node('server')
# service名, service型, 実行される関数
service = rospy.Service('service', string, callback)

rospy.spin()

File: catkin_ws/src/cpp_service/src/server_str.cpp
#include<ros/ros.h>
#include<cpp_service/string.h>

// コールバック関数
bool CallBack(cpp_service::string::Request &req, cpp_service::string::Response &res)
{
    ROS_INFO_STREAM("request:" << " " << req.x);
    res.y = "World!";
    return true;
}

int main(int argc, char **argv)
{
    // ノードの初期化
    ros::init(argc, argv, "str_server");
    // ノードハンドルの作成
    ros::NodeHandle nh;
    // service名, 実行される関数
    ros::ServiceServer srv = nh.advertiseService("string", &CallBack);
    ROS_INFO_STREAM("Service ready!");
    // 無限ループ的な？
    ros::spin();
}